docs/reference/scripting/access-fields.asciidoc

```diff
@@ -0,0 +1,86 @@
+[[script-fields-api]]
+== Access fields in a document with the `field` API
+++++
+<titleabbrev>Access fields in a document</titleabbrev>
+++++
+
+beta::["The `field` API is still in development and should be considered a beta feature. The API is subject to change and this iteration is likely not the final state.",{es-issue}78920]
+
+Use the `field` API to access document fields:
+
+[source,painless]
+----
+field('my_field').get()
+----
+
+This API fundamentally changes how you access documents in Painless. Previously,
+you had to access the `doc` map with the field name that you wanted to access:
+
+[source,painless]
+----
+doc['my_field'].value
+----
+
+Accessing document fields this way didn't handle missing values or missing
+mappings, which meant that your Painless scripts had to check that fields exist
+and create intermediary objects for handling such omissions.
```

Comment:
~and create intermediary objects for handling such omissions.~

and that values exist in order write robust scripts.

Reply:
To include your change, I'd like to rewrite this as:

>Accessing document fields this way didn't handle missing values or missing mappings, which meant that to write robust Painless scripts, you needed to include logic to check that both fields and values exist.